src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -198,25 +200,36 @@ struct PyModuleDef pythonmonkey =
   PythonMonkeyMethods
 };
 
+PyObject *SpiderMonkeyError = NULL;
+
 PyMODINIT_FUNC PyInit_pythonmonkey(void)
 {
   PyDateTime_IMPORT;
 
-  if (!JS_Init())
+  if (!JS_Init()) {
+    PyErr_SetString(SpiderMonkeyError, "Spidermonkey could not be initialized.");
     return NULL;
+  }
+    
 
   cx = JS_NewContext(JS::DefaultHeapMaxBytes);
-  if (!cx)
+  if (!cx) {
+    PyErr_SetString(SpiderMonkeyError, "Spidermonkey could not create a JS context.");
     return NULL;
+  }
 
-  if (!JS::InitSelfHostedCode(cx))
+  if (!JS::InitSelfHostedCode(cx)) {
+    PyErr_SetString(SpiderMonkeyError, "Spidermonkey could not initialize self-hosted code.");
     return NULL;
+  }
 
   JS::RealmOptions options;
   static JSClass globalClass = {"global", JSCLASS_GLOBAL_FLAGS, &JS::DefaultGlobalClassOps};
   global = new JS::RootedObject(cx, JS_NewGlobalObject(cx, &globalClass, nullptr, JS::FireOnNewGlobalHook, options));
-  if (!global)
+  if (!global) {
+    PyErr_SetString(SpiderMonkeyError, "Spidermonkey could not create a global object.");
     return NULL;
+  }
 
   Py_AtExit(cleanup);
```

Comment:
Need to register the cleanup function before any early exists, otherwise it would segfault
```
mozilla::detail::MutexImpl::~MutexImpl: pthread_mutex_destroy failed: Device or resource busy
Segmentation fault (core dumped)
```

Reply:
```suggestion
```